components/hab/src/main.rs

```diff
@@ -1312,6 +1322,26 @@ fn org_param_or_env(m: &ArgMatches) -> Result<String> {
     }
 }
 
+/// Check to see if a package idenifier or
+///
+fn ident_or_origin(m: &ArgMatches) -> ListingType {
```

Comment:
The name doesn't quite capture the functionality now.

Stepping back a bit, I wonder what it would be like to express this by implementing `From<&ArgMatches>` for `ListingType`, and if that might be a pattern we could use elsewhere in the CLI.

Reply:
That actually works pretty well.  I think the end logical conclusion for real grouping of implementation would be that the {valid_XXX} and the CLI `@arg` definition should be all together.  But that might be taking it a bit far